apm-lambda-extension/extension/process_metrics.go

```diff
@@ -0,0 +1,274 @@
+type StringMapItem struct {
+	// Key is the map item's key.
+	Key string
+
+	// Value is the map item's value.
+	Value string
+}
+
+// MetricLabel is a name/value pair for labeling metrics.
+type MetricLabel struct {
+	// Name is the label name.
+	Name string
+
+	// Value is the label value.
+	Value string
+}
+
+type MetricsContainer struct {
+	Metrics *Metrics `json:"metricset"`
+}
+
+// Metrics holds a set of metric samples, with an optional set of labels.
+type Metrics struct {
+	// Timestamp holds the time at which the metric samples were taken.
+	Timestamp int64 `json:"timestamp"`
+
+	// Transaction optionally holds the name and type of transactions
+	// with which these metrics are associated.
+	Transaction MetricsTransaction `json:"transaction,omitempty"`
+
+	// Span optionally holds the type and subtype of the spans with
+	// which these metrics are associated.
+	Span MetricsSpan `json:"span,omitempty"`
+
+	// Labels holds a set of labels associated with the metrics.
+	// The labels apply uniformly to all metric samples in the set.
+	//
+	// NOTE(axw) the schema calls the field "tags", but we use
+	// "labels" for agent-internal consistency. Labels aligns better
+	// with the common schema, anyway.
+	Labels StringMap `json:"tags,omitempty"`
+
+	// Samples holds a map of metric samples, keyed by metric name.
+	Samples map[string]Metric `json:"samples"`
+}
+
+// MetricsTransaction holds transaction identifiers for metrics.
+type MetricsTransaction struct {
+	Type string `json:"type,omitempty"`
+	Name string `json:"name,omitempty"`
+}
+
+// MetricsSpan holds span identifiers for metrics.
+type MetricsSpan struct {
+	Type    string `json:"type,omitempty"`
+	Subtype string `json:"subtype,omitempty"`
+}
+
+// Metric holds metric values.
+type Metric struct {
+	Type string `json:"type,omitempty"`
+	// Value holds the metric value.
+	Value float64 `json:"value"`
+	// Buckets holds the metric bucket values.
+	Values []float64 `json:"values,omitempty"`
+	// Count holds the metric observation count for the bucket.
+	Counts []uint64 `json:"counts,omitempty"`
+}
+
+// Add adds a metric with the given name, labels, and value,
+// The labels are expected to be sorted lexicographically.
+func (m *Metrics) Add(name string, labels []MetricLabel, value float64) {
+	m.addMetric(name, labels, Metric{Value: value})
+}
+
+// AddHistogram adds a histogram metric with the given name, labels, counts,
+// and values. The labels are expected to be sorted lexicographically, and
+// bucket values provided in ascending order.
+func (m *Metrics) AddHistogram(name string, labels []MetricLabel, values []float64, counts []uint64) {
+	m.addMetric(name, labels, Metric{Values: values, Counts: counts, Type: "histogram"})
+}
+
+// Simplified version of https://github.com/elastic/apm-agent-go/blob/675e8398c7fe546f9fd169bef971b9ccfbcdc71f/metrics.go#L89
+func (m *Metrics) addMetric(name string, labels []MetricLabel, metric Metric) {
+
+	var modelLabels StringMap
```

Comment:
Labels are not relevant given the current complexity required.

Reply:
Solved in [89772ef](https://github.com/elastic/apm-aws-lambda/pull/127/commits/89772eff89575d28155297d2cefab72ac0684593)